salt/utils/gitfs.py

```diff
@@ -528,6 +527,40 @@ def _get_per_saltenv(tgt_env):
                     else:
                         return None
 
+                if self.opts.get("__git_pillar", False) is True:
+                    if self.branch == "__env__":
+                        if hasattr(self, "all_saltenvs"):
+                            pillarenv = (
+                                self.opts.get("pillarenv")
+                                or self.opts.get("saltenv")
+                                or "base"
+                            )
+                            if tgt_env == pillarenv:
+                                return self.all_saltenvs
+                            else:
+                                return None
+                        else:
+                            if tgt_env == "base":
+                                return self.base
+                            else:
+                                return tgt_env
+                    elif self.env:
+                        if tgt_env == self.env:
+                            return self.branch
+                        else:
+                            return None
+                    else:
+                        if tgt_env == "base":
+                            if self.branch == self.base:
+                                return self.branch
+                            else:
+                                return None
+                        else:
+                            if tgt_env == self.branch:
+                                return self.branch
+                            else:
+                                return None
```

Comment:
If this check were extracted to an internal function, it would be easier to test.